### ISSUE TITLE
[6.1.0+] We copy the content + content from erased cells

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#6516]
### Description

In `6.1.0` we have introduced a regression (after [copyPaste epic](https://github.com/handsontable/handsontable/issues/4931)) when while pasting content from a merged cell we also copy the content from erased cells (erased by merge).

### Steps to reproduce

1. Select cell `A1` and copy its content
2. Paste it to `A3`

**Result in `6.0.1`**
We copy only the content of the cell.
![old](https://user-images.githubusercontent.com/10757813/69965078-c57bac80-1513-11ea-90e7-d30964396cc7.gif)

**Result in `6.1.0+`**
We copy the content + content from erased cells (erased by merge).
![new](https://user-images.githubusercontent.com/10757813/69965086-c7de0680-1513-11ea-8221-b6bb8b91e4d5.gif)

### Additional information
* `modifyCopyableRange` does not change
* `beforeCopy` changes its 2nd parameter
![image](https://user-images.githubusercontent.com/10757813/69965459-9a458d00-1514-11ea-9050-b9bd821a8259.png)
`{startRow: 2, startCol: 0, endRow: 3, endCol: 0} `
(`6.1.0+`) https://jsfiddle.net/s4tz0o9y/
`{startRow: 2, startCol: 0, endRow: 2, endCol: 0} `
(`6.0.1`) https://jsfiddle.net/at96gf4k/5/

* the `beforeCopy` has been changed in this commit https://github.com/handsontable/handsontable/commit/a9bdde3d6bcf6a5f3159c4a76ef097f6a136bb19

### Demo

https://jsfiddle.net/at96gf4k/4/ (`6.1.0` - broken)
https://jsfiddle.net/at96gf4k/3/ (`6.0.1` - working)

### Your environment
* Handsontable version: 6.1.0+
* Browser Name and version: Chrome 78
* Operating System: Windows 10

**Comment by AMBudnik** [comment]
**Works the same** in 8.0.0-beta2-rev14 https://jsfiddle.net/AMBudnik/be2ycwhL/

**Comment by pnowak** [comment]
This issue is still replicable using v8.0.0. Here's an updated demo - https://jsfiddle.net/x804d5wq/

**Comment by wszymanski** [comment]
Issue #5390 introduced big change in a way how the `CopyPaste` plugin works. Change mentioned within the issue pushes HOT's behavior in the right direction (data copied from merged cells looks properly). 
However, there is no information about merge and it definitely looks like regression. When https://github.com/handsontable/handsontable/issues/7094 will be done everything should be good.

We discussed with @wojciechczerniak about some ideas which tried to fix worse UX (i.e. change in `modifyCopyableRange` for selected, single merged cell), but they had the wrong consequences for other plugins (i.e. the `Autofill` plugin) and we decided that mentioned feature is essential.
